src/App.tsx

```diff
@@ -1,14 +1,80 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect, useMemo,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { SortBy } from './types/SortOption';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortOption, setSortOption] = useState(SortBy.All);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [activeTodoId, setActiveTodoId] = useState(0);
+
+  useEffect(() => {
+    const getTodosFromServer = async () => {
+      try {
+        const todosFromServer = await getTodos();
+
+        setTodos(todosFromServer);
+      } catch {
+        setError(true);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    getTodosFromServer();
+  }, []);
+
+  const getVisibleTodos = (
+    todosFilter: Todo[],
+    sortOptionFilter: SortBy,
+    queryFilter: string,
+  ): Todo[] => {
+    let filteredTodos = todos;
+
+    switch (sortOptionFilter) {
+      case SortBy.Active:
+        filteredTodos = todosFilter.filter(todo => !todo.completed);
+        break;
+
+      case SortBy.Completed:
+        filteredTodos = todosFilter.filter(todo => todo.completed);
+        break;
+
+      default:
+        break;
+    }
+
+    return filteredTodos.filter(todo => (
+      todo.title.toLowerCase().includes(queryFilter.toLowerCase())
+    ));
+  };
+
+  const activeTodo = useMemo(() => (
+    todos.find(todo => todo.id === activeTodoId)
+  ), [activeTodoId]);
```

Comment:
not sure about using useMemo here, since find method is not that hard to perform, we probably waste more resources to wrap it in useMemo